app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt

```diff
@@ -0,0 +1,125 @@
+package org.fossasia.openevent.general.settings
+
+import android.app.AlertDialog
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.os.Bundle
+import android.support.customtabs.CustomTabsIntent
+import android.support.v4.content.ContextCompat
+import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat
+import android.support.v7.preference.Preference
+import android.view.*
+import java.util.prefs.PreferenceChangeEvent
+import java.util.prefs.PreferenceChangeListener
+import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.BuildConfig
+import org.fossasia.openevent.general.MainActivity
+import org.fossasia.openevent.general.utils.nullToEmpty
+import org.koin.android.architecture.ext.viewModel
+
+class SettingsFragment : PreferenceFragmentCompat(), PreferenceChangeListener {
+    private var email: String? = null
+    val EMAIL: String = "EMAIL"
+    private val settingsViewModel by viewModel<SettingsFragmentViewModel>()
+
+    override fun preferenceChange(evt: PreferenceChangeEvent?) {
+        preferenceChange(evt)
+    }
+
+    override fun onCreatePreferencesFix(savedInstanceState: Bundle?, rootKey: String?) {
+        // Load the preferences from an XML resource
+        setPreferencesFromResource(R.xml.settings, rootKey)
+
+        val activity = activity as? MainActivity
```

Comment:
Never cast as specific activities, cast as generic action bar or support acitivity

Reply:
We are taking this approach in every other fragment, I will create an issue :+1: